application/frontend/controllers/UserController.php

```diff
@@ -23,14 +55,14 @@ public function actionCreate(): User
         return $user;
     }
 
-    public function actionUpdatePassword(string $employeeId): User
+    public function actionUpdatePassword($employeeId)
     {
-        $user = User::findOne([
-            'employee_id' => $employeeId
-        ]);
+        $user = User::findOne(['employee_id' => $employeeId]);
 
         if ($user === null) {
-            throw new NotFoundHttpException();
+            Yii::$app->response->statusCode = 204;
```

Comment:
This may be one of those situations where my push to use `204` (instead of `404`) for Not Found makes our API less intuitive. I think common practice would suggest that getting a `204` back (from a `PUT` call to update a password) would indicate success. 

Since this is following the spec we've decided on, it's probably best to leave it as `204` for now... as you said, @wcjr, we can change it later if we need to. `</second-guessing>`